app/src/main/java/com/li/musicmp3/data/source/local/MusicLocalDataSource.kt

```diff
@@ -0,0 +1,35 @@
+package com.li.musicmp3.data.source.local
+
+import android.content.Context
+import android.provider.MediaStore
+import com.li.musicmp3.data.model.Song
+import com.li.musicmp3.data.source.MusicDataSource
+import com.li.musicmp3.data.source.local.ultil.LoadMusicTask
+
+class MusicLocalDataSource(private val context: Context) : MusicDataSource {
+
+    override fun getSong(callBack: OnLoadMusicCallBack) {
+        LoadMusicTask(callBack) { getSongFromDevice() }.execute()
+    }
+
+    private fun getSongFromDevice(): List<Song> {
+        val resolver = context.contentResolver
+        val songs = ArrayList<Song>()
```

Comment:
prefer abstract than detail 
```suggestion
        val songs = mutableListOf<Song>()
```